wagtail/wagtailcore/tests/test_blocks.py

```diff
@@ -175,6 +175,39 @@ def test_validate_non_required_richtext_block(self):
         self.assertIsInstance(result, RichText)
         self.assertEqual(result.source, '')
 
+    def test_widget_field_richtext_block(self):
+        from wagtail.wagtailcore.fields import RichTextArea
+        custom_widget = RichTextArea(editor_config={'plugins': {'halloheadings': {'formatBlocks': ['p', 'h2']}}})
+        block = blocks.RichTextBlock(widget=custom_widget)
+        self.assertEqual(block.field_options.get('widget'), custom_widget)
+
+    def test_widget_custom_widget_richtext_block(self):
+        from wagtail.wagtailcore.fields import BaseTextAreaWidget
+
+        class CustomTextArea(BaseTextAreaWidget):
+            def get_panel(self):
+                pass
+        custom_widget = CustomTextArea(editor_config={'custom': {'testix': {'formatBlocks': ['p', 'h2']}}})
+        block = blocks.RichTextBlock(widget=custom_widget)
+
+        self.assertIsInstance(block.field_options.get('widget'), CustomTextArea)
+        self.assertIsInstance(block.field.widget, CustomTextArea)
```

Comment:
This test we can replace with this one:

```python
    def test_widget_custom_widget_richtext_block(self):
        custom_widget = RichTextArea(editor_config={'custom': {'testix': {'formatBlocks': ['p', 'h2']}}})
        block = blocks.RichTextBlock(widget=custom_widget)

        self.assertEqual(block.field.widget, custom_widget)
```

We don't need to define new widget class to cover changes in code: we just want to pass widget to `RichTextBlock`. Test fails even if `block.field.widget` will instance of `RichTextArea`, but not ours instance.